src/dos/cdrom_image.cpp

```diff
@@ -558,33 +870,38 @@ bool CDROM_Interface_Image::LoadCueSheet(char *cuefile)
 			|| command == "PERFORMER" || command == "POSTGAP" || command == "REM"
 			|| command == "SONGWRITER" || command == "TITLE" || command == "") success = true;
 		// failure
-		else success = false;
-
+		else {
+			delete track.file;
+			track.file = NULL;
+			success = false;
+		}
 		if (!success) return false;
 	}
 	// add last track
 	if (!AddTrack(track, shift, prestart, totalPregap, currPregap)) return false;
-	
+
 	// add leadout track
 	track.number++;
 	track.attr = 0;//sync with load iso
 	track.start = 0;
 	track.length = 0;
 	track.file = NULL;
-	if(!AddTrack(track, shift, 0, totalPregap, 0)) return false;
+	if(!AddTrack(track, shift, -1, totalPregap, 0)) return false;
```

Comment:
While touching this line, let's fix the formatting ;) (space between `if` and `(`). Also, return will look better on a new line.

Reply:
Fixed all of these in cdrom_image, and brought all the dangling returns under the conditional.
Doesn't look as elegant in some places, but it's at least consistent and some of those returns are far more obvious than being dangling out in the far-right of the text.
Fixed in 859cee72